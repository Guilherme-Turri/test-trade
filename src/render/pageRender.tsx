import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};