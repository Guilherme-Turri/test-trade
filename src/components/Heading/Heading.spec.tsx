import { componentRender } from "../../render/componentRender";
import { screen } from '@testing-library/react'
import { Heading } from ".";
describe('<Heading />', () => {

  it('should render on screen with Capital Letters', () => {
    componentRender(<Heading uppercase={true} big={false} as='h1'>test</Heading>)
    const msg = screen.getByText('test')
    expect(msg).toHaveStyleRule('text-transform', 'uppercase');
  })
  it('should render on screen with no Capital Letters', () => {
    componentRender(<Heading uppercase={false} big={false} as='h1'>Test</Heading>)
    const msg = screen.getByText('Test')
    expect(msg).toHaveStyleRule('text-transform', 'none');
  })
  it('should render on screen with small size', () => {
    componentRender(<Heading uppercase={false} big={false} as='h1'>Test</Heading>)
    const msg = screen.getByText('Test')
    expect(msg).toHaveStyleRule('font-size', '20px');
  })
  it('should render on screen with big size', () => {
    componentRender(<Heading uppercase={false} big={true} as='h1'>Test</Heading>)
    const msg = screen.getByText('Test')
    expect(msg).toHaveStyleRule('font-size', '60px');
  })
  it('should render on screen as h2', () => {
    componentRender(<Heading uppercase={false} big={true} as='h2'>Test</Heading>)
    const msg = screen.getByRole('heading', { level: 2 })
    expect(msg).toBeInTheDocument();
  })
  it('should render on screen as h3', () => {
    const { container } = componentRender(<Heading uppercase={false} big={true} as='h3'>Test</Heading>)
    const msg = screen.getByRole('heading', { level: 3 })
    expect(msg).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
})
