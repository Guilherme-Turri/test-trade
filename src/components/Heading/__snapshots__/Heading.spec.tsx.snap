// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render on screen as h3 1`] = `
.c0 {
  text-transform: none;
  color: #00c555;
  font-size: 60px;
  font-family: 'Bruno Ace SC',cursive;
}

<div>
  <h3
    class="c0"
  >
    Test
  </h3>
</div>
`;
